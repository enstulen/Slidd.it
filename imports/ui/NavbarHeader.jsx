import React, { Component, PropTypes } from 'react';
import { FlowRouter } from 'meteor/kadira:flow-router';

import Lecture from './Lecture.jsx';

export default class NavbarHeader extends Component {

  handleButtonPress(event) {
    // Go to the main page
    event.preventDefault();
    FlowRouter.go('/');
  }
  renderLectures() {
    return this.props.lectures.map((lecture) => (
      <Lecture key={lecture._id} lecture={lecture} />
    ));
  }

  render() {
    return (
      <div>
        <nav className="navbar navbar-default">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
                <span className="sr-only">Toggle navigation</span>
                <span className="icon-bar" />
                <span className="icon-bar" />
                <span className="icon-bar" />
              </button>
              <button className="invisibleButton" onClick={this.handleButtonPress}>
                <img className="img-responsive" width="180" alt="Slidd.it" src="/logo3.png" />
              </button>
            </div>
            <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
              <ul className="nav navbar-nav">
                <li><a href="/">Home</a></li>
                <li className="dropdown">
                  <a className="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Active lectures <span className="caret" /></a>
                  <ul className="dropdown-menu" aria-labelledby="dropdownMenu1">
                    {this.renderLectures()}
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}

// // // // // PropTypes // // // // //

NavbarHeader.propTypes = {
  lectures: PropTypes.array.isRequired,
};
