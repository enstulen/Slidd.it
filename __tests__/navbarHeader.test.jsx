import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import NavbarHeader from '../imports/ui/NavbarHeader.jsx';

describe('NavbarHeader component', () => {
  it('Renders correctly', () => {
    const rendered = renderer.create(
      <NavbarHeader lectures={[]} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Correct length', () => {

    const wrapper = shallow(<NavbarHeader lectures={[]} />);
    expect(wrapper.find('.container-fluid').length).toEqual(1);
    expect(wrapper.find('.navbar-default').length).toEqual(1);
    expect(wrapper.find('.collapsed').length).toEqual(1);
    expect(wrapper.find('.icon-bar').length).toEqual(3);
    expect(wrapper.find('.invisibleButton').length).toEqual(1);
    expect(wrapper.find('.img-responsive').length).toEqual(1);
    expect(wrapper.find('.navbar-collapse').length).toEqual(1);
    expect(wrapper.find('.navbar-nav').length).toEqual(1);
    expect(wrapper.find('.dropdown-toggle').length).toEqual(1);
    expect(wrapper.find('.dropdown-menu').length).toEqual(1);
    expect(wrapper.find('.liButton').length).toEqual(1);

  });
});
