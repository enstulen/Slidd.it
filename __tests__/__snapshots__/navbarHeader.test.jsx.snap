// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarHeader component Renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <button
          aria-expanded="false"
          className="navbar-toggle collapsed"
          data-target="#bs-example-navbar-collapse-1"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
        <button
          className="invisibleButton"
          onClick={[Function]}
        >
          <img
            alt="Slidd.it"
            className="img-responsive"
            src="/logo3.png"
            width="180"
          />
        </button>
      </div>
      <div
        className="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1"
      >
        <ul
          className="nav navbar-nav"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="dropdown"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              className="dropdown-toggle"
              data-toggle="dropdown"
              role="button"
            >
              Active lectures 
              <span
                className="caret"
              />
            </a>
            <ul
              aria-labelledby="dropdownMenu1"
              className="dropdown-menu"
            >
              <li>
                <button
                  className="liButton"
                  onClick={[Function]}
                >
                  Example class
                </button>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
